### ISSUE TITLE
identity function in createAction only handles the first parameter

### ISSUE THREAD

**Issue by killermonk** [acdlite/redux-actions#68]
The `createAction` function returns an action creator with the signature `function(...args)`. However, if you do not pass an `actionCreator` method to the `createAction` function, then the "`identity`" function is used.

The `identity` function is this: `(t) => t`, which does not do an argument spread and, since it is called with function.apply, it only handles the first argument passed in to the action creator, even though the action creator is designed around supporting n arguments.

**Comment by yangmillstheory** [comment]
What should the behavior be for the default function when N arguments are passed?

The returned `actionHandler` is responsible for building an FSA ([relevant code here](https://github.com/acdlite/redux-actions/blob/master/src/createAction.js#L9)) - it seems like the only reasonable thing to do when N arguments are passed is to return the first argument?

If you're calling an action creator with N arguments, it seems like that signature would imply/necessitate a custom action creator.

**Comment by yangmillstheory** [comment]
TL; DR I'm not sure there's an issue here.

**Comment by killermonk** [comment]
Sorry for my delay. I guess what I really want is something like this for identity:
`function(...arguments) { return arguments; }`

It just seemed off to me when reading through to code to see something allowing for n arguments, then ignoring them by default instead of just returning back those arguments.

**Comment by yangmillstheory** [comment]
Hm interesting. 

But in the case of one argument `'arg'`, would expect `action.payload` to be `['arg']`?

**Comment by killermonk** [comment]
Easy fix: `function(...args) { return args.length === 1 ? args[0] : args; }`

If I pass one argument, I'd just expect that argument. If I pass multiple arguments I'd expect an array.

In my use, though, if I'm passing arguments to the action, I also use a custom action creator with it. They are generally in the form of `(arg1, arg2) => ({arg1, arg2})`, so that I get a single object with the key being the name of the argument.

All in all, if you disagree with my opinion on how an identity should work, it's worth it just to close the ticket. This mostly just threw me off when I first started using this library.

**Comment by calvinnwq** [comment]
I personally do `(arg1, arg2) => ({ arg1, arg2 })` if i have arguments to pass.
Even if it were a single argument, I'd still do `(arg1) => ({ arg1 })` just to keep it consistent and make it more readable in the reducers, especially in larger apps.

e.g.
```javascript
// action
const actions = {
  entry: {
    add: createAction('ADD_ENTRY', (entry) => ({ entry })),
  }
};
// reducer
[actions.entry.add]: (state, payload) => ({
    ...state,
    entry: [
      ...state.entry,
      payload.entry,
    ],
  }),
```

Since I pass my own payloadCreator all the time, I don't encounter the problem as discussed here.

Also, since JavaScript does not have support for named parameters like in some other languages e.g. python where you can specify `functionName(arg1=1, arg2=2)`, you can pass an object literal as your only argument.

```javascript
const addEntry = createAction('ADD_ENTRY'),
const entry = { name: 'entry name' };
addEntry({ entry }); // same as addEntry({ entry: entry });
```

***

I don't really agree with passing multiple arguments as an array as payload, since you'd have to again assume the order of the arguments passed which makes code less readable and less flexible.

But, not passing all arguments is bad as well as it restricts by default, so I'd say the solution mentioned by @yangmillstheory above is sufficient.

**Comment by yangmillstheory** [comment]
@calvinnwq I like the pattern of named multiple arguments using a POJO. Definitely readable and more intentional even in the case of one argument.

**Comment by timche** [comment]
Whenever I need more than 1 argument to pass, I also use an object instead, so it is clear what the reducer will receive. Don't forget, `payload` can be any type of value.

**Comment by yangmillstheory** [comment]
@timche I'm OK with this approach (encouraging named parameters if I understand correctly), and it's something I use at work. 

Come to think of it, I haven't found great use cases for payload creators yet (I just use the identity with plain object payloads), and separately am interested to hear of any.

**Comment by yangmillstheory** [comment]
Closing this as it's not clear as there's a good and readable alternative (using a plain JS object).
